### PR TITLE
[ML] Add NLP inference configs to the inference processor docs

### PR DESCRIPTION
**Pull request by davidkyle**
### Preview

* https://elasticsearch_82320.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/inference-processor.html
* https://elasticsearch_82320.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/start-trained-model-deployment.html

### CODE REVIEW

**Comment by lcawl** [comment]
Should we also change the introduction in the "Inference processor" page, since it references only "data frame analytics models"?

File: docs/reference/ingest/processors/inference.asciidoc [hunk 1]
```diff
@@ -81,13 +81,243 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-classification
...
+=====
+`bert`::::
+(Optional, object)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert]
++
+.Properties of bert
+[%collapsible%open]
+=======
+
+`truncate`::::::
```
Comment (lcawl) [inline docs/reference/ingest/processors/inference.asciidoc:242]:
```suggestion
`truncate`::::
```

File: docs/reference/ingest/processors/inference.asciidoc [hunk 2]
```diff
@@ -81,13 +81,243 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-classification
...
+
+`mpnet`::::
+(Optional, object)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-mpnet]
++
+.Properties of mpnet
+[%collapsible%open]
+=======
+
+`truncate`::::::
```
Comment (lcawl) [inline docs/reference/ingest/processors/inference.asciidoc:255]:
```suggestion
`truncate`::::
```

File: docs/reference/ingest/processors/inference.asciidoc [hunk 3]
```diff
@@ -81,13 +81,243 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-classification
...
+=====
+`bert`::::
+(Optional, object)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert]
++
+.Properties of bert
+[%collapsible%open]
+=======
+
+`truncate`::::::
```
Comment (lcawl) [inline docs/reference/ingest/processors/inference.asciidoc:292]:
```suggestion
`truncate`::::
```

File: docs/reference/ingest/processors/inference.asciidoc [hunk 4]
```diff
@@ -81,13 +81,243 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-classification
...
+
+`mpnet`::::
+(Optional, object)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-mpnet]
++
+.Properties of mpnet
+[%collapsible%open]
+=======
+
+`truncate`::::::
```
Comment (lcawl) [inline docs/reference/ingest/processors/inference.asciidoc:305]:
```suggestion
`truncate`::::
```

File: docs/reference/ml/trained-models/apis/start-trained-model-deployment.asciidoc [hunk 5]
```diff
@@ -24,7 +24,9 @@ Requires the `manage_ml` cluster privilege. This privilege is included in the
 == {api-description-title}
 
 Currently only `pytorch` models are supported for deployment. When deployed,
-the model attempts allocation to every machine learning node.
+the model attempts allocation to every machine learning node. Once deployed
+the model can be used by the <<inference-processor,{infer-cap} processor>>
+in an ingest pipeline or directly in <<infer-trained-model-deployment>> API.
```
Comment (lcawl) [inline docs/reference/ml/trained-models/apis/start-trained-model-deployment.asciidoc:29]:
```suggestion
in an ingest pipeline or directly in the <<infer-trained-model-deployment>> API.
```

**Comment by davidkyle** [comment]
Thanks @lcawl I updated the introduction in https://github.com/elastic/elasticsearch/pull/82320/commits/4bd912d34ac3acbac374fadc4c2aa00615094a3e

File: docs/reference/ingest/processors/inference.asciidoc [hunk 6]
```diff
@@ -53,41 +167,156 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-results-field-
 (Optional, integer)
 include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-regression-num-top-feature-importance-values]
 
+[[inference-processor-text-classification-opt]]
```
Comment (szabosteve) [inline docs/reference/ingest/processors/inference.asciidoc:170]:
```suggestion
[discrete]
[[inference-processor-text-classification-opt]]
```

**Review by lcawl — APPROVED** [review]
LGTM, thanks!

**Pull request merged**